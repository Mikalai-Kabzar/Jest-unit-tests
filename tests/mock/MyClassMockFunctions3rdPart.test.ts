// MyClassGetMockName.test.ts
import MyClassGetMockName from './MyClassGetMockName';

describe('MyClassGetMockName', () => {
  it('should match snapshot for multiplyBy method with specific argument', () => {
    // Arrange
    const instance = new MyClassGetMockName(5);
    const mockMultiplyBy = jest.fn();

    // Replace the original method with the mock
    instance.multiplyBy = mockMultiplyBy;

    // Act
    instance.multiplyBy(3);

    // Assert
    expect(mockMultiplyBy).toHaveBeenCalledWith(3);
    expect(mockMultiplyBy).toMatchSnapshot();
  });

  it('should match snapshot for multiplyBy method with different argument', () => {
    // Arrange
    const instance = new MyClassGetMockName(5);
    const mockMultiplyBy = jest.fn();

    // Replace the original method with the mock
    instance.multiplyBy = mockMultiplyBy;

    // Act
    instance.multiplyBy(4);

    // Assert
    expect(mockMultiplyBy).toHaveBeenCalledWith(4);
    expect(mockMultiplyBy).toMatchSnapshot();
  });

  it('should match snapshot for getValue method', () => {
    // Arrange
    const instance = new MyClassGetMockName(10);
    const mockGetValue = jest.fn(() => 42);

    // Replace the original method with the mock
    instance.getValue = mockGetValue;

    // Act
    const result = instance.getValue();

    // Assert
    expect(result).toMatchSnapshot();
  });

  it('should match snapshot for multiplyBy method with mockImplementation', () => {
    // Arrange
    const instance = new MyClassGetMockName(5);
    const mockMultiplyBy = jest.fn().mockImplementation((factor: number) => factor * 2);

    // Replace the original method with the mock
    instance.multiplyBy = mockMultiplyBy;

    // Act
    const result = instance.multiplyBy(3);

    // Assert
    expect(result).toMatchSnapshot();
  });

  it('should match snapshot for multiplyBy method with mockReturnValue', () => {
    // Arrange
    const instance = new MyClassGetMockName(5);
    const mockMultiplyBy = jest.fn().mockReturnValue(42);

    // Replace the original method with the mock
    instance.multiplyBy = mockMultiplyBy;

    // Act
    const result = instance.multiplyBy(3);

    // Assert
    expect(result).toMatchSnapshot();
  });
});
