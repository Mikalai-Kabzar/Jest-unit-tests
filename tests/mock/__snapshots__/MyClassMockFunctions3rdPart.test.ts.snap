// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyClassGetMockName should match snapshot for getValue method 1`] = `42`;

exports[`MyClassGetMockName should match snapshot for multiplyBy method with different argument 1`] = `
[MockFunction] {
  "calls": [
    [
      4,
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`MyClassGetMockName should match snapshot for multiplyBy method with mockImplementation 1`] = `6`;

exports[`MyClassGetMockName should match snapshot for multiplyBy method with mockReturnValue 1`] = `42`;

exports[`MyClassGetMockName should match snapshot for multiplyBy method with specific argument 1`] = `
[MockFunction] {
  "calls": [
    [
      3,
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
